test/helpers/kubectl.go

```diff
@@ -795,6 +797,119 @@ func (kub *Kubectl) GetCiliumPodOnNode(namespace string, node string) (string, e
 	return res.Output().String(), nil
 }
 
+// isPodFinished return true if the pod has finish up running.
+func (kub *Kubectl) isPodFinished(ns, podName string) (bool, error) {
+	pod, err := kub.CoreV1().Pods(ns).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	switch pod.Status.Phase {
+	case v1.PodFailed, v1.PodSucceeded:
+		return true, nil
+	}
+	return false, nil
+}
+
+// WaitForPodFinish waits for the given pod in the given namespace until the pod
+// is finish running or until the given context deadline is reached.
+func (kub *Kubectl) WaitForPodFinish(ctx context.Context, ns, podName string) error {
+	for {
+		finish, err := kub.isPodFinished(ns, podName)
+		if err != nil {
+			return err
+		}
+		select {
+		case <-ctx.Done():
```

Comment:
We seem to be duplicating this code over and over again, can we have a generic helper that invokes a `func() (success bool, err error)` and waits for `success || err`?

Reply:
Maybe using `helpers.WithTimeout()`?